src/pybind/mgr/rbd_support/module.py

```diff
@@ -80,6 +80,21 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self.task = TaskHandler(self)
         self.trash_purge_schedule = TrashPurgeScheduleHandler(self)
 
+        # Previous scheduler handler versions incorrectly stored the
+        # global config in the localized module option. Check the config is
+        # here and fix it.
+        for handler in (self.mirror_snapshot_schedule, self.trash_purge_schedule):
```

Comment:
Why not do this in `Schedules::__init__`?  Then it wouldn't need to be a loop and schedule implementation quirks wouldn't leak to common rbd_support setup code.

Reply:
> Initially, I wanted to add this in Schedules::**init**. But I didn't do that since the Schedule's init gets called too during scheduler handler's load().

Ah, yeah, I keep forgetting about that...

> But I figured that my current change will be simple enough for now, and easy to remove in a later version when we no longer need to keep doing this.

I don't like how this implementation detail is spilled into another file.  Have you considered adding static (class) attribute so it's done only once but still in `Schedules`?

Reply:
I could only think of implementing it as a static method. Let me know if the latest version is better. If the static method doesn't look good then we can just move it out into just a plain function in schedule.py . Thanks!

Reply:
No longer need to do this. Taking the approach describe here, https://github.com/ceph/ceph/pull/49261#discussion_r1046665725